kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveDriver.java

```diff
@@ -159,6 +183,48 @@ public <K, V> Materialized<K, V, WindowStore<Bytes, byte[]>> materialized(
         .withLoggingEnabled(CHANGELOG_CONFIG);
   }
 
+  @Override
+  public <K, V> Materialized<K, V, KeyValueStore<Bytes, byte[]>> globalMaterialized(
+      final String name
+  ) {
+    return Materialized.<K, V>as(globalKv(name))
+        .withCachingDisabled();
+  }
+
+  @Override
+  public Admin getAdmin(final Map<String, Object> config) {
```

Comment:
This and all the below needs to be provided to the user as a `KafkaClientSupplier` rather than as individual callbacks. They'll do something like this:

```
final ResponsiveKafkaClientSupplier clientSupplier = responsiveDriver.getKafkaClientSupplier();
final KafkaStreams streams = new KafkaStreams(topology, config, clientSupplier);
```

Reply:
@ableegoldman I made `ResponsiveDriver` implement `KafkaClientSupplier` so they can just pas that right in :)

Reply:
Oh...why? Just seems rather sneaky and unintuitive, was there a reason to do it this way?

Reply:
based on our discussions, I assume there will be a time where the ResponsiveDriver needs to know which clients it handed out - this makes that dependency clear. Maybe I'm thinking too far ahead though and we can figure out that problem when we get there and for now we can just make it provide a supplier.

Reply:
I mean I'm totally on board with the ResponsiveDriver maintaining a connection to the client supplier, I just think it's a confusing API to present our poor users with. 

(Hope "sneaky and unintuitive" didn't come off as too harsh haha)

Reply:
> (Hope "sneaky and unintuitive" didn't come off as too harsh haha)

haha, you're saying sneaky isn't a compliment? 😉 just kidding, I see where you're coming from